tests/flow/test_index_delete.py

```diff
@@ -116,3 +116,39 @@ def test05_drop_index_during_population(self):
                 indicies = list_indicies(g).result_set
                 self.env.assertEquals(len(indicies), 0)
 
+    def test06_reset_order(self):
+        """Tests that the reset order is correct, i.e., that the reading ops are
+        reset before the writing ops (otherwise we write while a read-lock is
+        held --> deadlock)."""
+
+        # clear the db
+        self.env.flush()
+        g = Graph(self.env.getConnection(), GRAPH_ID)
+
+        # create data
+        g.query(
+            """
+            UNWIND range(1, 1000) AS x
+            CREATE (:X {uid: toString(x)})-[:R]->(y:Y {v: x})
+            """
+        )
```

Comment:
not sure why we need 1000 nodes, should 1 be enough ?
`CREATE (:X {uid: toString(x)})-[:R]->(y:Y {v: x})`